pkg/scanner/request.go

```diff
@@ -19,10 +20,17 @@ func WithStartBlock(blockHeight uint32) ScanRequestOption {
 	}
 }
 
+func WithPersistent() ScanRequestOption {
+	return func(req *ScanRequest) {
+		req.IsPersistent = true
+	}
+}
+
 func newScanRequest(options ...ScanRequestOption) *ScanRequest {
 	req := &ScanRequest{
-		Item:        nil,
-		StartHeight: 0,
+		Item:         nil,
+		StartHeight:  0,
+		IsPersistent: false,
 	}
```

Comment:
No need to specify the "zero" for every field, this can just be:

```suggestion
        req := &ScanRequest{}
```